pagination/src/main/java/com/cerner/beadledom/pagination/OffsetPaginationModule.java

```diff
@@ -0,0 +1,141 @@
+package com.cerner.beadledom.pagination;
+
+import com.cerner.beadledom.pagination.parameters.LimitParameter;
+import com.cerner.beadledom.pagination.parameters.OffsetParameter;
+import com.google.inject.AbstractModule;
+import com.google.inject.Provides;
+import com.google.inject.name.Named;
+import javax.inject.Singleton;
+
+/**
+ * Module to provide the necessary configurations for using Offset Based Pagination.
+ *
+ * <p>Module provides the configurations needed to use offset based pagination. This includes
+ * the {@link OffsetPaginatedListLinksWriterInterceptor} as well as default values for the limit and
+ * offset fields and their respective field names. The default values for the limit, offset, and
+ * their respective field names can be configured using one of the provided constructors.
+ *
+ * @author Will Pruyn
+ * @author Ian Kottman
+ * @since 3.1
+ */
+public class OffsetPaginationModule extends AbstractModule {
+  private final Integer maxLimit;
+  private final Integer defaultLimit;
+  private final String defaultLimitFieldName;
+  private final Long defaultOffset;
+  private final String defaultOffsetFieldName;
+
+  private static final Integer BEADLEDOM_DEFAULT_LIMIT = 20;
+  private static final Integer BEADLEDOM_DEFAULT_MAX_LIMIT = 100;
+  private static final String BEADLEDOM_DEFAULT_LIMIT_FIELD_NAME = "limit";
+  private static final Long BEADLEDOM_DEFAULT_OFFSET = 0L;
+  private static final String BEADLEDOM_DEFAULT_OFFSET_FIELD_NAME = "offset";
+
+  /**
+   * Default constructor that sets the configurable fields using the Beadledom defaults.
+   */
+  public OffsetPaginationModule() {
+    this.defaultLimit = BEADLEDOM_DEFAULT_LIMIT;
+    this.maxLimit = BEADLEDOM_DEFAULT_MAX_LIMIT;
+    this.defaultOffset = BEADLEDOM_DEFAULT_OFFSET;
+    this.defaultLimitFieldName = BEADLEDOM_DEFAULT_LIMIT_FIELD_NAME;
+    this.defaultOffsetFieldName = BEADLEDOM_DEFAULT_OFFSET_FIELD_NAME;
+  }
+
+  /**
+   * Constructor that enables setting of default values for the offset and limit fields.
+   * Default field names will be set using Beadledom defaults.
+   * @param defaultLimit the default value for the limit field; 20 if null.
+   * @param maxLimit the default value for the limit field; 100 if null.
+   * @param defaultOffset the default value for the offset field; 0 if null.
+   */
+  public OffsetPaginationModule(Integer defaultLimit, Integer maxLimit, Long defaultOffset) {
```

Comment:
Dont know if you guys thought about using autovalue for this class also so that we could have a builder instead of a bunch of constructors? 
or if you want to keep the constructors have you thought about instead of setting the params in each one just passing the params along to one so you dont have to have all the default log checking in each one.